Sources/Login/AuthenticationCoordinator.swift

```diff
@@ -67,6 +67,9 @@ final class AuthenticationCoordinator: NSObject,
     
     func handleUniversalLink(_ url: URL) {
         do {
+            if let loginCoordinator = parentCoordinator as? LoginCoordinator {
+                loginCoordinator.introViewController?.enableIntroButton()
+            }
```

Comment:
I've made `LoginCoordinator.introViewController` an internal property so that we can call it here in the `AuthenticationCoordinator`. I did this rather than passing it in to make it more simplistic, happy to change if there is a better way to do this.
This is done here because when selecting a universal link, which is fed through to this method, the below call to `AppAuthSession.finalise(edirectURL:)` immediately dismisses the authentication modal before triggering the above `AppAuthSession.performLoginFlow(configuration:)`. This means that the button loading will remain until `finish()` is called on line 36. This isn't terrible UX and could be fine where we're going; presenting a loading screen on top of this screen until the work to call the `/token` endpoint is complete and we finish this coordinator successfully or with an error. Though, for the meantime, I have implemented this call here to ensure the button is reenabled.